### PR TITLE
Adds error overlay to display runtime errors

### PR DESCRIPTION
**Pull request by madsnedergaard**
## Description

Ensure more runtime errors are shown in an error modal to make life easier while developing.

### Before
Screenshot 2022-11-02 at 20 21 25

### After

Screenshot 2022-11-02 at 20 21 47

### CODE REVIEW

File: web/src/utils/errorOverlay.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  // Don't open outside vite environment
+  if (!ErrorOverlay) {
+    return;
+  }
+  console.log(error);
+  const overlay = new ErrorOverlay(error);
+  document.body.append(overlay);
+};
+
+export default function showErrorsInOverlay() {
```
Comment (tonypls) [inline web/src/utils/errorOverlay.ts:14]: Maybe better named as enableErrorsInOverlay so it's easy to diffrentiate it from the showErrorOverlay

**Review by tonypls — APPROVED** [review]
👍 Optional change :D

File: web/src/utils/errorOverlay.ts (hunk 1 above)
Comment (madsnedergaard) [inline web/src/utils/errorOverlay.ts:14]:
```suggestion
export default function enableErrorsInOverlay() {
```

File: web/src/main.tsx [hunk 2]
```diff
@@ -6,9 +6,14 @@ import { registerSW } from 'virtual:pwa-register';
 import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
 import './index.css';
 import { REFETCH_INTERVAL_MS } from 'api/helpers';
+import showErrorsInOverlay from 'utils/errorOverlay';
```
Comment (madsnedergaard) [inline web/src/main.tsx:9]:
```suggestion
import enableErrorsInOverlay from 'utils/errorOverlay';
```

File: web/src/main.tsx [hunk 3]
```diff
@@ -6,9 +6,14 @@ import { registerSW } from 'virtual:pwa-register';
...
 
 registerSW();
 
+if (import.meta.env.DEV) {
+  showErrorsInOverlay();
```
Comment (madsnedergaard) [inline web/src/main.tsx:14]:
```suggestion
  enableErrorsInOverlay();
```

File: web/src/utils/errorOverlay.ts (hunk 1 above)
Comment (madsnedergaard) [reply web/src/utils/errorOverlay.ts]: Good point :)

**Pull request merged**